src/executables/util_cs.c

```diff
@@ -3994,6 +3994,798 @@ tde (UTIL_FUNCTION_ARG * arg)
+	}
+
+      sprintf (*sql + sql_length, "%d", i_data);
+      break;
+    case 1:
+      ptr = or_unpack_int64 (ptr, &b_data);
+
+      error = check_sql_memory (sql, sql_length + max_decimal_digits_len, max_sql_size);
+      if (error != NO_ERROR)
+	{
+	  return error;
+	}
+
+      sprintf (*sql + sql_length, "%lld", b_data);
+      break;
+    case 2:
+      ptr = or_unpack_float (ptr, &f_data);
+      error = check_sql_memory (sql, sql_length + max_decimal_digits_len, max_sql_size);
+      if (error != NO_ERROR)
+	{
+	  return error;
+	}
+
+      sprintf (*sql + sql_length, "%f", f_data);
+      break;
+    case 3:
+      ptr = or_unpack_double (ptr, &d_data);
+      error = check_sql_memory (sql, sql_length + max_decimal_digits_len, max_sql_size);
+      if (error != NO_ERROR)
+	{
+	  return error;
+	}
+
+      sprintf (*sql + sql_length, "%lf", d_data);
+      break;
+    case 4:
+      ptr = or_unpack_short (ptr, &sh_data);
+      error = check_sql_memory (sql, sql_length + max_decimal_digits_len, max_sql_size);
+      if (error != NO_ERROR)
+	{
+	  return error;
+	}
+
+      sprintf (*sql + sql_length, "%d", sh_data);
+      break;
+    case 5:
+      ptr = or_unpack_string_nocopy (ptr, &s_data);
+      error = check_sql_memory (sql, sql_length + strlen (s_data) + 3, max_sql_size);
+      if (error != NO_ERROR)
+	{
+	  return error;
+	}
+
+      if (IS_QOUTES_NEEDED (type))
+	{
+	  sprintf (*sql + sql_length, "\'%s\'", s_data);
+	}
+      else
+	{
+	  sprintf (*sql + sql_length, "%s", s_data);
+	}
+      break;
+    case 7:
+      ptr = or_unpack_string_nocopy (ptr, &s_data);
+
+      error = check_sql_memory (sql, sql_length + strlen (s_data) + 3, max_sql_size);
+      if (error != NO_ERROR)
+	{
+	  return error;
+	}
+
+      if (IS_QOUTES_NEEDED (type))
+	{
+	  sprintf (*sql + sql_length, "\'%s\'", s_data);
+	}
+      else
+	{
+	  sprintf (*sql + sql_length, "%s", s_data);
+	}
+
+      break;
+    default:
+      break;
+    }
+
+  *data = ptr;
+
+  return NO_ERROR;
+}
+
+static void
+print_flashback_detail (int trid, char *user, char *flashback, char *original, FILE * outfp)
+{
+  fprintf (outfp, "[TRANSACTION ID] %d \n", trid);
+  fprintf (outfp, "[USER]           %s \n", user);
+  fprintf (outfp, "[ORIGINAL]       %s \n", original);
+  fprintf (outfp, "[FLASHBACK]      %s \n", flashback);
+}
+
+static int
+print_flashback_update (char **loginfo, int trid, char *user, const char *classname, bool is_detail, FILE * outfp)
+{
+  MOP classop;
+  SM_CLASS *class_;
+  SM_ATTRIBUTE *attr;
+
+  int error = NO_ERROR;
+
+  char *sql = NULL;
+  int max_sql_size = 1024;
+
+  char *cond_sql = NULL;
+  int max_cond_sql_size = 1024;
+
+  char *original_sql = NULL;
+  int max_original_size = 1024;
+
+  char *start_ptr;
+  char *ptr;
+
+  int num_change_col;
+  int change_index;
+
+  int num_cond_col;
+  int cond_index;
+
+  int i = 0;
+
+  sql = (char *) malloc (max_sql_size);
+  if (sql == NULL)
+    {
+      error = ER_OUT_OF_VIRTUAL_MEMORY;
```

Comment:
er_set(). and check others too.

Reply:
or you think it doesn't need?

Reply:
Oh.. I will add 
`util_log_write_errid (MSGCAT_UTIL_GENERIC_NO_MEM)`

I will do other error setting and message setting, too